### PR TITLE
Add more ArrayConverter tests and cleanup code

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ArrayConverter.cs [hunk 1]
```diff
@@ -35,26 +30,20 @@ public override object ConvertTo(ITypeDescriptorContext context, CultureInfo cul
         /// </summary>
         public override PropertyDescriptorCollection GetProperties(ITypeDescriptorContext context, object value, Attribute[] attributes)
         {
-            if (value == null)
+            if (!(value is Array valueArray))
             {
-                return null;
+                return new PropertyDescriptorCollection(null);
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ArrayConverter.cs]: Why is it ok to change this from returning null to returning new PropertyDescriptorCollection(null) for a null input?

Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ArrayConverter.cs]: Firstly, this code was added in .Net core. Full framework throws NRE. Secondly, by changing this to return an empty list we actually match the behaviour of other converters that implement getproperties

Comment (stephentoub) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ArrayConverter.cs]: Thanks, but this semantic change isn't cleaning up code nor adding tests. Please revert it. If you want to try to make such a change, please do it in a dedicated PR specifically for that, with an explanation for why it's important.

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**